compiler/next/include/chpl/queries/Context-detail.h

```diff
@@ -64,73 +65,37 @@ template<typename ResultType, typename... ArgTs> class QueryMapResult;
 class QueryMapBase;
 template<typename ResultType, typename... ArgTs> class QueryMap;
 
-// define a hash function for std::tuple since the standard doesn't
-// include one.
-template<typename T>
-static std::size_t queryArgsHashOne(const T& v) {
-  std::hash<T> hasher;
-  return hasher(v);
-}
-
-template<size_t I, typename... Ts>
-static std::size_t queryArgsHashImpl(const std::tuple<Ts...>& tuple) {
-  size_t myhash = queryArgsHashOne(std::get<I>(tuple));
-  if (I + 1 < sizeof...(Ts)) {
-    size_t resthash =
-      queryArgsHashImpl<(I + 1 < sizeof... (Ts) ? I + 1 : I)>(tuple);
-    return chpl::hash_combine(myhash, resthash);
-  } else {
-    return myhash;
-  }
+template<typename TUP, size_t... I>
+static inline bool queryArgsEqualsImpl(const TUP& lhs, const TUP& rhs, std::index_sequence<I...>)
+{
+  // TODO: C++17 fold expression
+  // return(... && (std::get<I>(lhs) == std::get<I>(rhs)));
+
+  // C++14 code
+  // This expands the ret = ret && std::get<I>(lhs) == std::get<I>(rhs)
+  // assignments into a brace initializer list, evaluating the elements of
+  // I in left-to-right order. The value of ret is true iff all elements
```

Comment:
evaluating the elements of I -> evaluating the elements of TUP ?

Reply:
I guess I could have phrased it that way.  But `I` is the parameter pack variable which is being expanded, and then `lhs` and `rhs` (both tuples) are indexed through that.